utility/pcfg_loader.cpp

```diff
@@ -0,0 +1,194 @@
+#include "pcfg_loader.h"
+#include <core/input/input_event.h>
+#include <core/config/engine.h>
+#include <core/os/keyboard.h>
+#include <core/io/compression.h>
+#include <core/os/file_access.h>
+#include <core/io/marshalls.h>
+#include <core/variant/variant_parser.h>
+#include "bytecode/bytecode_base.h"
+#include "utility/variant_writer_compat.h"
+
+Error ProjectConfigLoader::load_cfb(const String path, const uint32_t ver_major, const uint32_t ver_minor){
+
+    cfb_path = path;
+	Error err;
+	FileAccess *f = FileAccess::open(path, FileAccess::READ, &err);
+	ERR_FAIL_COND_V_MSG(!f, err, "Could not open " + path);
+	err = _load_settings_binary(f, path, ver_major);
+	memdelete(f);
+    return err;
+}
+
+Error ProjectConfigLoader::save_cfb(const String dir, const uint32_t ver_major, const uint32_t ver_minor){
+	String file;
+	if (ver_major > 2) {
+		file = "project.godot";
+	} else {
+		file = "engine.cfg";
+	}
+
+	return save_custom(dir.plus_file(file), ver_major, ver_minor);
+}
+
+Error ProjectConfigLoader::_load_settings_binary(FileAccess * f, const String &p_path, uint32_t ver_major) {
+
+	Error err;
+	uint8_t hdr[4];
+	f->get_buffer(hdr, 4);
+	if (hdr[0] != 'E' || hdr[1] != 'C' || hdr[2] != 'F' || hdr[3] != 'G') {
+
+		ERR_FAIL_V_MSG(ERR_FILE_CORRUPT, "Corrupted header in binary project.binary (not ECFG).");
+	}
+
+	uint32_t count = f->get_32();
+
+	for (uint32_t i = 0; i < count; i++) {
+
+		uint32_t slen = f->get_32();
+		CharString cs;
+		cs.resize(slen + 1);
+		cs[slen] = 0;
+		f->get_buffer((uint8_t *)cs.ptr(), slen);
+		String key;
+		key.parse_utf8(cs.ptr());
+
+		uint32_t vlen = f->get_32();
+		Vector<uint8_t> d;
+		d.resize(vlen);
+		f->get_buffer(d.ptrw(), vlen);
+		Variant value;
+		if (ver_major == 4){
+			err = decode_variant(value, d.ptr(), d.size(), NULL, true);
+		} else if (ver_major == 3){
+			err = GDScriptDecomp::decode_variant_3(value, d.ptr(), d.size(), NULL, true);
+		} else if (ver_major == 2){
+			err = GDScriptDecomp::decode_variant_2(value, d.ptr(), d.size(), NULL, true);
+
+		}
```

Comment:
```suggestion
		} else {
			err = ERR_INVALID_PARAMETER; //or some other error code
		}
```
`err` is uninitialized if `ver_major` is not one of the supported.